### PR TITLE
Np 2890 all scenarios

### PR DESCRIPTION
**Pull request by axthosarouris**
Focus on the feature files and the glue code. Changes in the CristinDataGenerator are basically the change of all non-static methods to static and making some methods public.

### CODE REVIEW

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 1]
```diff
@@ -21,60 +38,80 @@
...
-import static no.unit.nva.cristin.mapper.CristinObject.MAIN_CATEGORY_FIELD;
-import static no.unit.nva.cristin.mapper.CristinObject.PUBLICATION_OWNER_FIELD;
-import static no.unit.nva.cristin.mapper.CristinObject.SECONDARY_CATEGORY_FIELD;
-import static no.unit.nva.hamcrest.DoesNotHaveEmptyValues.doesNotHaveEmptyValues;
-import static no.unit.nva.hamcrest.DoesNotHaveEmptyValues.doesNotHaveEmptyValuesIgnoringFields;
-import static nva.commons.core.attempt.Try.attempt;
-import static org.hamcrest.MatcherAssert.assertThat;
-
-public class CristinDataGenerator {
+public final class CristinDataGenerator {
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:41]: most changes here are that all methods are static now and some have become public as well.,

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 2]
```diff
@@ -21,60 +38,80 @@
...
+                   .withFamilyName(randomString())
+                   .withAffiliations(randomAffiliations())
+                   .build();
     }
 
-    public String randomDataAsString() {
-        return randomObjects()
-                .map(this::toJsonString)
-                .collect(Collectors.joining(System.lineSeparator()));
+    public static List<CristinContributorsAffiliation> randomAffiliations() {
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:74]: method moved due to change of visibility

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 3]
```diff
@@ -21,60 +38,80 @@
...
+            case ANTHOLOGY:
+                return randomBookAnthology();
+            default:
+                break;
+        }
+        throw new IllegalStateException(
+            String.format("The secondary category %s is not covered", secondaryCategory));
+    }
+
+    public static <T> AwsEventBridgeEvent<FileContentsEvent<JsonNode>> toAwsEvent(T inputData) {
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:114]: moved methods due to change of visibility

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 4]
```diff
@@ -83,225 +120,211 @@ public static String randomString() {
...
     }
 
-    public String singleRandomObjectAsString() {
+    public static String singleRandomObjectAsString() {
         return newCristinObject(0).toJsonString();
     }
 
-    public static CristinContributorsAffiliation createAffiliation(CristinContributorRoleCode roleCode) {
-        return creatCristinContributorsAffiliation(roleCode);
+    public static JsonNode objectWithCustomMainCategory(String customMainCategory) throws JsonProcessingException {
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:139]: moved method

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 5]
```diff
@@ -83,225 +120,211 @@ public static String randomString() {
...
-                .withSubdepartmentIdentifier(threeDigitPositiveNumber())
-                .withDepartmentIdentifier(largeRandomNumber())
-                .withRoles(List.of(createRole(roleCode)))
-                .build();
+    private static int threeDigitPositiveNumber() {
+        return RANDOM.nextInt(1000);
     }
 
-    private CristinObject createRandomBookWithBookReportValues() {
+    private static CristinObject createRandomBookWithSpecifiedSecondaryCategory(
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:222]: mostly spacing

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java [hunk 6]
```diff
@@ -88,8 +87,7 @@ public void init() {
         s3Client = new FakeS3Client(new ConcurrentHashMap<>());
         handler = new CristinEntryEventConsumer(resourceService, s3Client);
         outputStream = new ByteArrayOutputStream();
-        cristinDataGenerator = new CristinDataGenerator();
-        super.testingData = cristinDataGenerator.singleRandomObjectAsString();
+        super.testingData = CristinDataGenerator.singleRandomObjectAsString();
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java:90]: cristinDataGenerator has now only static methods and there is no need for a  specific object

**Review by brinxmat — DISMISSED** [review]

**Pull request merged**